src/Compilers/VisualBasic/Portable/Semantics/TypeInference/TypeArgumentInference.vb

```diff
@@ -1408,11 +1408,12 @@ HandleAsAGeneralExpression:
 
 
                 Dim parameterElementTypes As ImmutableArray(Of TypeSymbol) = Nothing
-                If parameterType.TryGetElementTypesIfTupleOrCompatible(parameterElementTypes) Then
+                Dim argumentElementTypes As ImmutableArray(Of TypeSymbol) = Nothing
 
-                    Dim argumentElementTypes As ImmutableArray(Of TypeSymbol) = Nothing
-                    If Not argumentType.TryGetElementTypesIfTupleOrCompatible(argumentElementTypes) OrElse
-                            parameterElementTypes.Length <> argumentElementTypes.Length Then
+                If parameterType.GetNullableUnderlyingTypeOrSelf().TryGetElementTypesIfTupleOrCompatible(parameterElementTypes) AndAlso
+                   argumentType.TryGetElementTypesIfTupleOrCompatible(argumentElementTypes) Then
```

Comment:
It looks like we need special handling for that scenario as well. The code below that deals with digging through matching generic types on both sides doesn't do the right thing for Nullable as well.